### PR TITLE
Send donor name, email and address to PayPal Donations payment details

### PR DESCRIPTION
**Pull request by ravinderk**
## Description

The donor fills in personal details in the donation form (v2 and v3) and the PayPal Donations payment modal. This pull request resolves this issue and auto-fills available donor data in the PayPal Donations payment modal.

Note: Paypal tracks donor location. For this reason, the country field in the modal is always set to your country. If you are testing with a donor address other than your country, then in modal, selecting the donor country manually will auto-fill the state field (if any). You can read about this behavior in detail in the [Slack thread](https://lw.slack.com/archives/C04UT6UPERW/p1692056039090419)

## Affects

This pull request affects one-time and subscription donations with PayPal Donations.

## Testing Instructions

- [ ] Donor data like name and address auto-fill in the Paypal modal when donating with v2 donation forms
   - [ ] One time donation
   - [ ] Subscription
- [ ] Donor data like name and address auto-fill in the Paypal modal when donating with v3 donation forms
   - [ ] One time donation
   - [ ] Subscription

Note: PayPal processes transactions in a modal when donors subscribe to the subscription using the `Debit and Credit Card` smart button.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205769898178684
  - https://app.asana.com/0/0/1205318504721016

### CODE REVIEW

File: src/PaymentGateways/PayPalCommerce/AjaxRequestHandler.php [hunk 1]
```diff
@@ -250,11 +251,8 @@ public function createOrder()
                 'firstName' => $postData['give_first'],
                 'lastName' => $postData['give_last'],
                 'email' => $postData['give_email'],
-                'address' => $this->getDonorAddressFromPostedDataForPaypalOrder($postData),
-            ],
-            'application_context' => [
-                'shipping_preference' => 'NO_SHIPPING',
-            ],
+                'address' => $donorAddress,
```
Comment (glaubersilva) [inline src/PaymentGateways/PayPalCommerce/AjaxRequestHandler.php:254]: We need an unreleased tag for this change.

**Review by glaubersilva — CHANGES_REQUESTED** [review]
@ravinderk Nice work! I left a comment about just a minor change.

**Review by glaubersilva — APPROVED** [review]
@ravinderk Thanks for the changes, ready for QA!

**Review by matheuswd — APPROVED** [review]
Approved!

**Comment by ravinderk** [comment]
Merging is blocked query resolved. [Checking query details in the Slack](https://lw.slack.com/archives/C04UT6UPERW/p1698239889286709?thread_ts=1692056039.090419&cid=C04UT6UPERW).

**Comment by ravinderk** [comment]
@JasonTheAdams This pull request is ready to release. [Check the PayPal Team response for more details](https://lw.slack.com/archives/C04UT6UPERW/p1698678030124869?thread_ts=1692056039.090419&cid=C04UT6UPERW).

**Pull request merged**